### PR TITLE
Interaction system

### PR DESCRIPTION
**Pull request by kocosephia**

### CODE REVIEW

File: WhosAGoodBoy/Who's A Good Boy/Assets/Scripts/Interaction/Interactables/House.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class House : Interactable
```
Comment (nihakue) [inline WhosAGoodBoy/Who's A Good Boy/Assets/Scripts/Interaction/Interactables/House.cs]: Does C# / Unity have a concept of an interface / trait? I don't like hiding that House : MonoBehaivor but maybe that's standard. Also, what if House needs to implement other behavior? Does C# have multiple inheritance?

File: WhosAGoodBoy/Who's A Good Boy/Assets/Scripts/Extensions.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public static class Extensions{
+
+    public static Collider2D GetClosest(this Collider2D[] colliders, Vector3 relativeToHere)
```
Comment (nihakue) [inline WhosAGoodBoy/Who's A Good Boy/Assets/Scripts/Extensions.cs]: This is neat, are there limits to what you can write extension functions for in C#?

File: WhosAGoodBoy/Who's A Good Boy/Assets/Scripts/Interaction/Interactable.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Interactable : MonoBehaviour {
```
Comment (nihakue) [inline WhosAGoodBoy/Who's A Good Boy/Assets/Scripts/Interaction/Interactable.cs]: [nitpick] the class is called Interactable but the package is Interactables

File: WhosAGoodBoy/Who's A Good Boy/Assets/Scripts/Interaction/Interactables/House.cs (hunk 1 above)
Comment (kocosephia) [reply WhosAGoodBoy/Who's A Good Boy/Assets/Scripts/Interaction/Interactables/House.cs]: c# doesn't support multiple inheritance. It supports interfaces though.

File: WhosAGoodBoy/Who's A Good Boy/Assets/Scripts/Extensions.cs (hunk 2 above)
Comment (kocosephia) [reply WhosAGoodBoy/Who's A Good Boy/Assets/Scripts/Extensions.cs]: I don't think so. Any class can be extended

File: WhosAGoodBoy/Who's A Good Boy/Assets/Scripts/Interaction/Interactable.cs (hunk 3 above)
Comment (nihakue) [reply WhosAGoodBoy/Who's A Good Boy/Assets/Scripts/Interaction/Interactable.cs]: Sorry, the directory that this class is in

File: WhosAGoodBoy/Who's A Good Boy/Assets/Scripts/Interaction/Interactables/House.cs (hunk 1 above)
Comment (nihakue) [reply WhosAGoodBoy/Who's A Good Boy/Assets/Scripts/Interaction/Interactables/House.cs]: Does an Interactable interface work?

**Pull request merged**